pkg/controller/machine.go

```diff
@@ -470,7 +468,7 @@ func (c *controller) machineDelete(machine *v1alpha1.Machine, driver driver.Driv
 					c.targetCoreClient,
 					timeOutDuration, // TODO: Will need to configure timeout
 					nodeName,
-					-1,
+					int(timeOutDuration.Seconds()),
```

Comment:
What is it for ?

Reply:
I thought to cap the graceful termination of any pod by the drain timeout should help in overall drain option going through in the drain timeout. But i dunno if it works as expected.

I am hoping that if a pod is set to a graceful termination of `2Hours`, and our drain timeout is '5mins` we max would only give 5mins?

Reply:
I have reverted this change here - https://github.com/gardener/machine-controller-manager/pull/248/commits/7510f9ce74835b969614d94cb0cfa65f392becea#diff-d8287fe74b5273163c9f6b6c635ad912R475.

Reply:
Thanks.